### PR TITLE
[10.x] Add per-minute rate limiting test

### PR DESCRIPTION
**Pull request by timacdonald**
Wrote this test when I thought there was a bug in the rate limiter.

Turns out this was the bug: https://github.com/laravel/framework/pull/48497

But I figured it wouldn't hurt to have an extra test up in here. Never let a good test go to waste, I say.

Will rebase once we merge the other bugfix.

### CODE REVIEW

**Comment by timacdonald** [comment]
Note to self: check if retry-after ever reaches zero.

File: src/Illuminate/Routing/Middleware/ThrottleRequests.php [hunk 1]
```diff
@@ -287,8 +287,8 @@ protected function getHeaders($maxAttempts,
         ];
 
         if (! is_null($retryAfter)) {
-            $headers['Retry-After'] = $retryAfter;
-            $headers['X-RateLimit-Reset'] = $this->availableAt($retryAfter);
+            $headers['Retry-After'] = $retryAfter + 1;
+            $headers['X-RateLimit-Reset'] = $this->availableAt($retryAfter + 1);
```
Comment (crynobone) [inline src/Illuminate/Routing/Middleware/ThrottleRequests.php:291]: Changes to `Retry-After` looks logical but not sure about `X-RateLimit-Reset`

Comment (crynobone) [reply src/Illuminate/Routing/Middleware/ThrottleRequests.php:291]: > The GitHub API rate limit ensures that the API is fast and available for everyone. If you hit a rate limit, you should stop making requests **until after the time specified** by the x-ratelimit-reset header.

Comment (timacdonald) [reply src/Illuminate/Routing/Middleware/ThrottleRequests.php:291]: @crynobone the problem I see with that is we are showing a lie. You can't make requests after that moment in time - you have to wait until of the second.

The rate limit doesn't reset at the value in `X-RateLimit-Reset`. It resets at the value in `X-RateLimit-Reset` + 999 milliseconds.

I dunno. I struggle with this. I can see this one both ways.

**Pull request closed without merging**